app/widgets/Search/Search.php

```diff
@@ -19,6 +21,22 @@ public function load()
     {
         $this->addjs('search.js');
         $this->addcss('search.css');
+        $this->registerEvent('iqgateway_set_handle', 'onIqGatewaySet');
+    }
+
+    public function foundJid($prompt, $jid) {
+        $session = \Movim\Session::start();
```

Comment:
We need to store state somewhere, since this data comes in asynchronously from XMPP and there can be more than one.  The search results are rendered using `draw` and expect the entire list of results to be available for that.  Additionally, we do probably want to cache this for a short time since it's a remote lookup.

I'm not saying memory session is definitely right for it, but I'm curious what you would suggest as an alternative here in order to have `prepareSearch` know about all the items that have come in asynchronously?